process/caller.py

```diff
@@ -41,7 +43,36 @@ def check_agreement(
         :return: whether values agree or not
         :rtype: bool
         """
-        return np.allclose(previous, current, rtol=1.0e-6)
+        # Check for same shape: mfile length can change between iterations
+        if isinstance(previous, float) or previous.shape == current.shape:
+            return np.allclose(previous, current, rtol=1.0e-6, equal_nan=True)
+        else:
+            return False
+
+    @staticmethod
+    def check_mfile_agreement(
+        previous: dict[str, float], current: dict[str, float]
+    ) -> dict[str, list[float, float]]:
+        """Compare previous and current mfiles for agreement within a tolerance.
+
+        :param previous: variables and values from previous models evaluation
+        :type previous: dict[str,float]
+        :param current: variables and values from current models evaluation
+        :type current: dict[str,float]
+        :return: non-converged values
+        :rtype: dict[str, list[float, float]
+        """
+        nonconverged_vars = {}
+        for var in previous.keys():
+            if np.allclose(
+                previous[var], current.get(var, np.nan), rtol=1.0e-6, equal_nan=True
```

Comment:
I wonder whether having NaN's not equal each other will be helpful. Then, you get all the NaNs output in the terminal:

```
Variable      Previous value    Current value
----------  ----------------  ---------------
coe                      nan              nan
```

Reply:
Yes, I agree it's worth highlighting `nan`s, but this table is for non-idempotent values; the `nan` is idempotent in this case.